stellargraph/connector/neo4j/graph.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018-2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__all__ = ["Neo4jStellarGraph"]
+
+import numpy as np
+import scipy.sparse as sps
+import pandas as pd
+from ...core.experimental import experimental
+
+
+@experimental(reason="the class is not tested", issues=[1578])
+class Neo4jStellarGraph:
+    """
+    Neo4jStellarGraph class for graph machine learning on graphs stored in
+    a Neo4j database.
+
+    This class communicates with Neo4j via a p2neo.Graph connected to the graph
+    database of interest and contains functions to query the graph data necessary
+    for machine learning.
+
+    Args:
+        graph (py2neo.Graph): a py2neo.Graph connected to a Neo4j graph database.
+        is_directed (bool, optional): If True, the data represents a
+            directed multigraph, otherwise an undirected multigraph.
+    """
+
+    def __init__(self, graph_db, is_directed=False):
+
+        self.graph_db = graph_db
+        self._is_directed = is_directed
+
+    def nodes(self):
+        """
+        Obtains the collection of nodes in the graph.
+
+        Returns:
+            The node IDs of all the nodes in the graph.
+        """
+        node_ids_query = f"""
+            MATCH (n)
+            RETURN n.ID as node_id
+            """
+
+        result = self.graph_db.run(node_ids_query)
+        data = result.data()
+        return [row["node_id"] for row in data]
```

Comment:
Tiny but this could be:

```suggestion
        return [row["node_id"] for row in result.data()]
```